### PR TITLE
RatingCalculationEnum is replaced by a RatingPoints entity

### PR DESCRIPTION
**Pull request by Maryna-511750**
# GreenCity

## Issue Link :clipboard:
[#7496](https://github.com/ita-social-projects/GreenCity/issues/7496)

## Added
* Added RatingPoints entity to handle the point allocation for different achievements.

## Changed
* Refactored existing logic to remove RatingCalculationEnum and replaced it with RatingPoints.
* Updated services, repositories, and methods related to rating calculation logic.
* Modified unit and integration tests to validate the new rating calculation approach.

## Deleted
* Removed the RatingCalculationEnum as it is no longer needed.

# PR Checklist Forms

_(to be filled out by PR submitter)_
- [ ] Code is up-to-date with the `dev` branch.
- [x] You've successfully built and run the tests locally.
- [x] There are new or updated unit tests validating the change.
- [x] JIRA/ Github Issue number & title in PR title (ISSUE-XXXX: Ticket title)
- [x] This template filled (above this section).
- [x] Sonar's report does not contain bugs, vulnerabilities, security issues, code smells ar duplication
- [x] `NEED_REVIEW` and `READY_FOR_REVIEW` labels are added.
- [x] All files reviewed before sending to reviewers

### CODE REVIEW

File: dao/src/main/java/greencity/entity/RatingPoints.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Entity
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+@Table(name = "rating_points")
+public class RatingPoints {
```
Comment (RomaMocherniuk) [inline dao/src/main/java/greencity/entity/RatingPoints.java:20]: please use `record`

File: service-api/src/main/java/greencity/dto/ratingstatistics/RatingPointsDto.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@AllArgsConstructor
+@NoArgsConstructor
+@Data
+@Builder
+public class RatingPointsDto {
```
Comment (RomaMocherniuk) [inline service-api/src/main/java/greencity/dto/ratingstatistics/RatingPointsDto.java:12]: please use `records` for all data clasees

File: service/src/main/java/greencity/achievement/AchievementCalculation.java [hunk 3]
```diff
@@ -114,7 +117,7 @@ private void saveAchievementToUser(UserVO userVO, AchievementCategoryVO ac, int
                 userAchievement.setHabit(habitRepo.findById(habitId).orElseThrow(() -> new NotFoundException(
                     ErrorMessage.HABIT_NOT_FOUND_BY_ID + habitId)));
             }
-            RatingCalculationEnum reason = RatingCalculationEnum.findByName(achievement.getTitle());
+            RatingPoints reason = ratingPointsRepo.findByNameOrThrow(achievement.getTitle());
```
Comment (RomaMocherniuk) [inline service/src/main/java/greencity/achievement/AchievementCalculation.java]: should the variable be remained?

File: dao/src/main/java/greencity/entity/RatingPoints.java (hunk 1 above)
Comment (Maryna-511750) [reply dao/src/main/java/greencity/entity/RatingPoints.java:20]: But it is an `Entity`. `Entity` can't be a `record`, can it?

File: service-api/src/main/java/greencity/dto/ratingstatistics/RatingPointsDto.java (hunk 2 above)
Comment (Maryna-511750) [reply service-api/src/main/java/greencity/dto/ratingstatistics/RatingPointsDto.java:12]: For this DTO requires a no-argument constructor, which is not supported by `records`.

File: service/src/main/java/greencity/achievement/AchievementCalculation.java (hunk 3 above)
Comment (Maryna-511750) [reply service/src/main/java/greencity/achievement/AchievementCalculation.java]: Not necessarily. I fixed it

**Review by KizerovDmitriy — APPROVED** [review]

**Review by K4sik — APPROVED** [review]

**Review by IliaWithHat — APPROVED** [review]

**Pull request merged**